Display status emojis in the LHN (including the user's own status)

perunt opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
This is the first part of a series of PRs to implement the UI for the new custom status feature.
It implements the Status option in the profile settings, and the screen to set a status.
As this feature is incomplete yet and build in iterations it will be hidden behind the `customStatus`  beta.



### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/23132
PROPOSAL: 


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
Add the customStatus beta to your account or simply [let this function](https://github.com/Expensify/App/blob/1c34c4bc90e043f20da8b94aac644dc2e2dc9b94/src/libs/Permissions.js#L101-L103) return true

Test Steps for Account A:

1. Log in using Account A.
2. Navigate to Settings.
3. Select the Status item.
4. Choose the desired emoji and accompanying text.
5. Press Save.
 - Note: By default, the status is set to expire 'by the end of the day,' which means it remains valid until tomorrow.
6. Return to the main dashboard.
7. Verify that the chosen emoji now appears next to your avatar.

Test Steps for Account B:
1. Log in using Account B.
2. View the chat list on the left panel.
3. Locate the chat associated with Account A.
4. Confirm that you can see Account A's status emoji next to their name.
5. (For Web Only) Hover over the status emoji.
 - Verify that a tooltip appears.
 - Confirm that the text within the tooltip matches what was set by Account A.

Additional Notes:
- Modifying the valid time duration for the status can only be done manually within the code.[ Refer to the provided link for the specifics](https://github.com/margelo/expensify-app-fork/blob/2303985708918ff984a4110f234415fa3e193a6d/src/pages/settings/Profile/CustomStatus/StatusPage.js#L47).

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/48593211/c558beb3-2017-45da-9ef8-418fa9ff1a7f


</details>

<details>
<summary>Mobile Web - Chrome</summary>

![photo_2023-08-15 12 43 54](https://github.com/Expensify/App/assets/48593211/e14e9383-978a-47e4-a6d2-f945e80cbc2f)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![Simulator Screen Shot - SE 16 2 - 2023-08-15 at 12 42 05](https://github.com/Expensify/App/assets/48593211/747e76e3-05f6-4eb4-9b08-dad74919536a)


</details>

<details>
<summary>Desktop</summary>


![Screenshot 2023-08-15 at 12 45 49](https://github.com/Expensify/App/assets/48593211/0e3014d0-3ce5-4cb6-84c1-b84ae7d7f932)

</details>

<details>
<summary>iOS</summary>

![Simulator Screen Shot - iPhone 14 Pro - 2023-08-15 at 12 59 58](https://github.com/Expensify/App/assets/48593211/9e43ea62-d681-45f3-bd02-718e78eb4000)



</details>

<details>
<summary>Android</summary>

![photo_2023-08-15 13 32 30](https://github.com/Expensify/App/assets/48593211/902a29e4-f2b3-47f7-9f2e-60ed2069edd0)


</details>

perunt committed:
add status details to personalDetailsSelector

perunt committed:
add status next to user name

perunt committed:
add status to sidebar link

perunt committed:
remove unused code

perunt committed:
add time in status

perunt committed:
Merge branch 'main' of https://github.com/margelo/expensify-app-fork into feat/##23132-display-status-lhn

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

perunt committed:
split status and profile buttons

perunt committed:
fix goBack on Status page

perunt committed:
add translation

perunt committed:
fix margin

perunt committed:
clean AvatarWithIndicator

perunt committed:
update isPlainDirectMessage

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

perunt committed:
add canUseCustomStatus to mocks permissions

perunt committed:
change translation

perunt committed:
return genericEditFailureMessage

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

burczu commented:
@perunt Please update Tests chapter in your Author Checklist - especially how to set this status be able to test clicking on it to show the Status Page...

burczu commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -124,6 +132,13 @@ function OptionRowLHN(props) {
+    const statusClearAfterDate = lodashGet(optionItem, 'status.clearAfter', '');
+    const formattedDate = DateUtils.getStatusUntilDate(statusClearAfterDate);
+    const statusContent = formattedDate ? `${statusText} (${formattedDate})` : statusText;
+    const isStatusVisible = Permissions.canUseCustomStatus(props.betas) && emojiCode && ReportUtils.isShowEmojiStatus(optionItem);
```
this line will throw an error because there is no `isShowEmojiStatus` in the `ReportUtils` module

perunt replied:
👍🏻

burczu commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -202,6 +217,14 @@ function OptionRowLHN(props) {
+                                                text={statusContent}
+                                                shiftVertical={-4}
+                                            >
+                                                <Text style={styles.ml1}>{`${emojiCode}`}</Text>
```
```suggestion
                                                <Text style={styles.ml1}>{emojiCode}</Text>
```

perunt replied:
👍🏻

burczu commented on src/pages/home/sidebar/SidebarLinks.js:
```diff
@@ -174,6 +171,93 @@ class SidebarLinks extends React.PureComponent {
+        Navigation.navigate(ROUTES.SETTINGS_STATUS);
+    }
+
+    renderAvatarWithIndicator() {
```
I would extract this method to the separate functional component

perunt replied:
👍🏻

burczu commented on src/pages/home/sidebar/SidebarLinks.js:
```diff
@@ -174,6 +171,93 @@ class SidebarLinks extends React.PureComponent {
+     * Render either a sign-in button or an avatar with optional status.
+     * @returns {React.Component}
+     */
+    renderSignInOrAvatarWithOptionalStatus() {
```
The same here - this should be a separate component used in the `render` method

perunt replied:
👍🏻

perunt committed:
create SignInOrAvatarWithOptionalStatus

perunt committed:
create PressableAvatarWithIndicator

perunt committed:
clean SidebarLinks

perunt commented:
@burczu  I updated the description and also added the [link](https://github.com/margelo/expensify-app-fork/blob/2303985708918ff984a4110f234415fa3e193a6d/src/pages/settings/Profile/CustomStatus/StatusPage.js#L47) to the method of how you can change the date manually
you can pass it using this formatting `moment(...).format('YYYY-MM-DD HH:mm:ss')`

perunt committed:
update isPlainDirectMessage

perunt committed:
update isStatusVisible

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

perunt commented:
Yep, it's an identical error `Invalid prop forwardedRef supplied to withOnyx(Component)`
The same error we have in the change language screen. Seem to be related to Onyx.
@stitesExpensify if status is a beta feature can we postpone fixing [this](https://github.com/Expensify/App/issues/22929) error? wdyt?

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

stitesExpensify commented:
Yes, we can skip that for now since it's already being worked on elsewhere. I've been OOO but will review this on monday

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

burczu commented on src/pages/home/sidebar/PressableAvatarWithIndicator.js:
```diff
@@ -0,0 +1,61 @@
+const propTypes = {
+    isCreateMenuOpen: PropTypes.bool,
+
+    currentUserPersonalDetails: personalDetailsPropType,
```
please add comments to propTypes

perunt replied:
👍🏻

burczu commented on src/pages/home/sidebar/SignInOrAvatarWithOptionalStatus.js:
```diff
@@ -0,0 +1,107 @@
+
+    isCreateMenuOpen: PropTypes.bool,
+
+    betas: PropTypes.arrayOf(PropTypes.string),
```
same here: missing comments

perunt replied:
👍🏻

burczu commented on src/pages/home/sidebar/PressableAvatarWithIndicator.js:
```diff
@@ -0,0 +1,61 @@
+}
+
+PressableAvatarWithIndicator.propTypes = propTypes;
+PressableAvatarWithIndicator.defaultProps = defaultProps;
```
missing `displayName`

burczu commented on src/pages/home/sidebar/SignInOrAvatarWithOptionalStatus.js:
```diff
@@ -0,0 +1,107 @@
+}
+
+SignInOrAvatarWithOptionalStatus.propTypes = propTypes;
+SignInOrAvatarWithOptionalStatus.defaultProps = defaultProps;
```
missing `displayName`

perunt replied:
👍🏻

burczu reviewed (commented):
added some additional minor comments, besides that we will be good to go, I think

burczu commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

perunt committed:
add comments

burczu commented on src/pages/settings/Profile/CustomStatus/StatusPage.js:
```diff
@@ -37,19 +37,19 @@ function StatusPage({draftStatus, currentUserPersonalDetails}) {
+        User.clearDraftCustomStatus();
+    };
+
+    const onNavigateBackToSettingsPage = useCallback(() => Navigation.goBack(ROUTES.SETTINGS_PROFILE, false, true), []);
```
based on Reviewer Checklist we should avoid naming event handlers like `onSomething` so please change this handler's name to something like `navigateBackToSettingsPage`

perunt replied:
👍🏻

burczu commented on src/pages/home/sidebar/SignInOrAvatarWithOptionalStatus.js:
```diff
@@ -0,0 +1,110 @@
+        Navigation.navigate(ROUTES.SETTINGS_STATUS);
+    }, [isCreateMenuOpen]);
+
+    if (Session.isAnonymousUser()) {
```
I also think that the JSX returned in these two if's (line 57 and 75) could be extracted to separate components, to separate concerns - what do you think?

perunt replied:
I haven't refactored all the components, but moving it outside makes sense. I'll update it. Thanks for pointing it up!

burczu reviewed (commented):
two additional comments after going through Reviewer Checklist

perunt committed:
create separate component for SignInButton

stitesExpensify commented on src/libs/ReportUtils.js:
```diff
@@ -2412,6 +2412,25 @@ function isIOUOwnedByCurrentUser(report, allReportsDict = null) {
+ * @param {Object} report (chatReport or iouReport)
+ * @returns {boolean}
+ */
+function isPlainDirectMessage(report) {
```
```suggestion
function isOneOnOneChat(report) {
```

stitesExpensify reviewed (commented):
Looking good, one small comment!

burczu commented on src/pages/home/sidebar/SignInOrAvatarWithOptionalStatus.js:
```diff
@@ -0,0 +1,95 @@
+    if (Session.isAnonymousUser()) {
+        return <SignInButton />;
+    }
+    if (emojiStatus) {
```
and what about this if? this is also a candidate to be refactored

perunt replied:
👍🏻

perunt committed:
rename isPlainDirectMessage to isOneOnOneChat

perunt committed:
a bit more optimizations

perunt committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/##23132-display-status-lhn

burczu reviewed (approved):
Reviewed and tested - I think we are good to go!

stitesExpensify commented:
Looking great so far! 2 things that I noticed:
1. The user's status is not being shown in the LHN:
<img width="1195" alt="2023-08-22_15-58-48" src="https://github.com/Expensify/App/assets/42391420/a7af17ad-2c41-4565-be2f-a2977ec919ae">
2. In @burczu testing videos, the status is showing up on the `#admins` channel which shouldn't happen. The status should only ever show up on DMs. I was not able to repro this so maybe it isn't happening anymore? But something i wanted to call out in case you have an idea how that could happen @perunt 

https://user-images.githubusercontent.com/7771498/262036263-7db7c2d3-20d6-4cdf-a603-dca8c07e4d46.mp4

perunt commented:
I added a check for that second issue. As for the first one, it's unrelated to focus mode like I thought. I can't get it to happen on my end. Would you mind sharing the test account details or that JSON report?

perunt committed:
add participantsList check

stitesExpensify commented:
Hmm there is nothing special about the accounts i'm using. @burczu are you able to see the emojis in `1` from my post above?

burczu commented:
@stitesExpensify @perunt Regarding first issue - I think it works correctly: the status icon shows up in LHN only next to items where the current user, that has this status set, is presented:

![Screenshot 2023-08-24 at 09 37 39](https://github.com/Expensify/App/assets/7771498/5a281e32-b5b1-4856-a9be-13e7f06eece9)

![Screenshot 2023-08-24 at 09 37 59](https://github.com/Expensify/App/assets/7771498/b3718099-c024-41e9-ac46-0977b4edfef7)

burczu commented on src/libs/ReportUtils.js:
```diff
@@ -2503,6 +2503,26 @@ function isIOUOwnedByCurrentUser(report, allReportsDict = null) {
+        !isTaskReport(report) && //
+        isDM(report) &&
+        !isIOUReport(report) &&
+        report.participantsList.length === 1
```
I don't this correctly fixes the issue with showing up status icon next to channels like `#admins` - in my case there is only one admin, and the icons keeps showing up

stitesExpensify commented:
> I think it works correctly: the status icon shows up in LHN only next to items where the current user, that has this status set, is presented:

Hmm something must just be weird on my end then.

perunt committed:
add one more check for isOneOnOneChat

stitesExpensify reviewed (approved):
LGTM!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6003310178) to staging by https://github.com/stitesExpensify in version: 1.3.58-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6019254807) to production by https://github.com/luacmartins in version: 1.3.58-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6019259194) to staging by https://github.com/stitesExpensify in version: 1.3.59-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6031542575) to production by https://github.com/luacmartins in version: 1.3.59-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

WoLewicki commented on src/pages/home/sidebar/SidebarLinks.js:
```diff
@@ -204,29 +184,7 @@ class SidebarLinks extends React.PureComponent {
-                            </OfflineWithFeedback>
-                        )}
-                    </PressableWithoutFeedback>
+                    <SignInOrAvatarWithOptionalStatus isCreateMenuOpen={this.props.isCreateMenuOpen} />
```
@perunt could you elaborate from where does `this.props.isCreateMenuOpen` comes? Or is it always `undefined`?

perunt replied:
Hmm, tbh I didn't dig where it comes from but now I can see that it was undefined 
I see that it was removed (`isCreateMenuOpen` and `onAvatarClick`) without cleaning it inside the component
![image](https://github.com/Expensify/App/assets/48593211/75701faa-f105-4c6f-b97f-b422bce50e8a)

WoLewicki replied:
Ok thanks for the input.

stitesExpensify commented:
I think that something broke here. I'm no longer seeing the status in LHN at all @perunt

perunt commented:
I saw recently the Moment was replaced, it could break it. Let me check

stitesExpensify commented:
<img width="1200" alt="2023-09-27_13-47-23" src="https://github.com/Expensify/App/assets/42391420/74008a58-9baf-4582-a98b-abbe35491c62">

shubham1206agra commented on src/pages/settings/Profile/CustomStatus/StatusPage.js:
```diff
@@ -37,19 +37,19 @@ function StatusPage({draftStatus, currentUserPersonalDetails}) {
+        User.clearDraftCustomStatus();
+    };
+
+    const navigateBackToSettingsPage = useCallback(() => Navigation.goBack(ROUTES.SETTINGS_PROFILE, false, true), []);
```
This change created this issue https://github.com/Expensify/App/issues/30509.

allroundexperts commented on src/pages/home/sidebar/SignInButton.js:
```diff
@@ -0,0 +1,33 @@
+        <PressableWithoutFeedback
+            accessibilityLabel={translate('sidebarScreen.buttonMySettings')}
+            accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}
+            onPress={Session.signOutAndRedirectToSignIn}
```
This caused https://github.com/Expensify/App/issues/34764 since `signOutAndRedirectToSignIn` got the first param as `true` always. It should have been `false` instead.
